truss/remote/baseten/core.py

```diff
@@ -47,18 +47,42 @@ def exists_model(api: BasetenApi, model_name: str) -> Optional[str]:
     return None
 
 
-def get_model_versions_info(api: BasetenApi, model_name) -> Tuple[str, dict]:
-    query_result = api.get_model(model_name)["model_version"]["oracle"]
+def get_model_versions_info(api: BasetenApi, model_name: ModelName) -> Tuple[str, List]:
+    query_result = api.get_model(model_name.value)["model"]
     return (query_result["id"], query_result["versions"])
 
 
+def get_model_versions_info_by_id(
+    api: BasetenApi, model_id: ModelId
+) -> Tuple[str, List]:
+    query_result = api.get_model_by_id(model_id.value)["model"]
+    return (query_result["id"], query_result["versions"])
+
+
+def get_dev_version_info_from_versions(versions: List[dict]) -> dict:
+    for version in versions:
+        if version["is_draft"] is True:
+            return version
+    raise ValueError("No development version found")
+
+
 def get_dev_version_info(api: BasetenApi, model_name: str) -> dict:
     model = api.get_model(model_name)
-    versions = model["model_version"]["oracle"]["versions"]
+    versions = model["model"]["versions"]
+    try:
+        return get_dev_version_info_from_versions(versions)
+    except ValueError:
+        raise ValueError(f"No development version found with model name: {model_name}")
+
+
+def get_prod_version_info_from_versions(versions: List[dict]) -> dict:
+    # Loop over versions instead of using the primary_version field because
+    # primary_version is set to the development version ID if no published
+    # models exist.
     for version in versions:
-        if version["is_draft"] is True:
+        if version["is_primary"] and not version["is_draft"]:
```

Comment:
I don't think it's possible for `is_primary` to be true for development models, but thought it'd be safer to check `is_draft` in case the `is_primary` definition ever changes